src/handwritten/_install_daqmx.py

```diff
@@ -0,0 +1,154 @@
+    """
+    try:
+       with urllib.request.urlopen(metadata_url) as url:
+           data = json.load(url)
+           windows_os_meta = data.get('Windows', [])
+       
+       for metadata in windows_os_meta:
+           location = metadata.get('Location')
+           version = metadata.get('Version')
+           if location and version:
+               return location, version
+               
+    except Exception as e:
+        print(f"Unable to get download url: {e}")
+    return None, None
+
+
+def get_platform_str():
+    """
+    get the platform name on which  
+    """
+    return platform.system().lower()
+
+
+def install_daqmx(download_url, temp_file):
+    try:
+        urllib.request.urlretrieve(download_url, temp_file)
+        subprocess.Popen([temp_file], shell=True)
+    except Exception as e:
+        print(f"Error installing NI-DAQmx: {e}")
+
+
+def ask_user_confirmation(question):
+    while True:
+        response = input(question + " (yes/no): ").strip().lower()
+        if response in ['yes', 'y']:
+            return True
+        elif response in ['no', 'n']:
+            return False
+        else:
+            print("Please enter 'yes' or 'no'.")
+
+def install_daqmx_software():
+    current_os = get_platform_str()
+
+    if current_os == "windows":
+        if is_admin():
+            daqmx_latest_url, daqmx_latest_version = get_url_and_version_for_windows_installer(META_DATA_URL)
+            if daqmx_latest_version == None:
+               return 0
+            daqmx_installed_version = check_daqmx_is_installed()
+            download_url, temp_file = daqmx_latest_url, get_temp_file_name()
+            if not daqmx_installed_version:
+               print(f"Installing NI-DAQmx version {daqmx_latest_version}")
+               install_daqmx(download_url, temp_file)
+
+            elif compare_version(daqmx_latest_version, daqmx_installed_version) > 0:
+                if download_url and temp_file:
+                    user_question = "Installed NI-DAQmx version is " +  daqmx_installed_version + ". Latest version available is " +  daqmx_latest_version + ". Do you want to upgrade?" 
+                    user_confirmation = ask_user_confirmation(user_question)
+                    if user_confirmation:
+                        print(f"Installing NI-DAQmx version {daqmx_latest_version}")
+                        install_daqmx(download_url, temp_file)
+                    else:
+                        print(f"Skipping Upgrade.")
+                    
+                else:
+                    print("Installation aborted due to missing download URL or temporary file.")
+            else:
+                print(f"Installed NI-DAQmx version {daqmx_installed_version} is up-to-date. Skipping installation.")
+        else:
+            print("Administrator privileges required. Installation aborted.")
+    else:
+        print("Unsupported OS detected.")
+
+
+if __name__ == "__main__":
+    install_daqmx_software()
```

Comment:
Either `_main` or `_install_daqmx` is ok with me.

(I don't think suffixing the function name with `_software` adds anything, unless it's necessary to avoid ambiguity, and I don't think that's the case.)